src/math/CurveFitting.cc

```diff
@@ -243,59 +240,80 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
-      #endif
-      ndarray::Array<T, 1, 1> D_A1_YFit;
-      D_A1_YFit = pfs::drp::stella::math::Poly(D_A1_XArr, 
-                                               D_A1_Coeffs_Out,
-                                               T(xRange[0]),
-                                               T(xRange[1]));
-
+                                                        keyWords,
+                                                        args);
+      LOGLS_DEBUG(_log, "PolyFit(D_A1_XArr, D_A1_YArr, I_Degree_In, keyWords, args) returned D_A1_Coeffs_Out = " << D_A1_Coeffs_Out);
+      LOGLS_DEBUG(_log, "yFit = " << *pYFit);
       ndarray::Array<T, 1, 1> D_A1_Temp = ndarray::allocate(D_A1_Y.size());
-      for (int pos = 0; pos < D_A1_Y.size(); ++pos)
-        D_A1_Temp[pos] = D_A1_Y[pos] - D_A1_YFit[pos];
+      auto itY = D_A1_YArr.begin();
+      auto itYFit = pYFit->begin();
+      for (auto itTemp = D_A1_Temp.begin(); itTemp < D_A1_Temp.end(); ++itTemp, ++itY, ++itYFit)
+        *itTemp = (*itY) - (*itYFit);
+      LOGLS_DEBUG(_log, "1. D_A1_Temp = " << D_A1_Temp);
       Eigen::Array<T, Eigen::Dynamic, 1> tempEArr = D_A1_Temp.asEigen();
       D_A1_Temp.asEigen() = tempEArr.pow(2) / T(D_A1_Y.size());
-//        D_A1_Temp.deep() = D_A1_Temp / D_A1_Y.size();
+      LOGLS_DEBUG(_log, "2. D_A1_Temp = " << D_A1_Temp);
       double D_SDev = double(sqrt(D_A1_Temp.asEigen().sum()));
-
-      D_A1_PolyRes.deep() = pfs::drp::stella::math::Poly(xNew, 
+      P_D_A1_YFit->deep() = pfs::drp::stella::math::Poly(D_A1_X_In,
                                                          D_A1_Coeffs_Out,
-                                                         T(xRange[0]),
-                                                         T(xRange[1]));
+                                                         xRange[0],
+                                                         xRange[1]);
+      LOGLS_DEBUG(_log, "P_D_A1_YFit = " << *P_D_A1_YFit);
       double D_Dev;
       D_A1_X.resize(0);
       D_A1_Y.resize(0);
-      D_A1_MeasureErrors.resize(0);
+      V_MeasureErrors.resize(0);
       I_A1_OrigPos.resize(0);
       P_I_A1_Rejected->resize(0);
       for (size_t i_pos=0; i_pos < D_A1_Y_In.getShape()[0]; i_pos++){
-        D_Dev = D_A1_Y_In[i_pos] - D_A1_PolyRes[i_pos];
+        D_Dev = D_A1_Y_In[i_pos] - (*P_D_A1_YFit)[i_pos];
+        LOGLS_DEBUG(_log, "i_pos = " << i_pos << ": D_Dev = " << D_Dev << ", D_SDev = " << D_SDev);
         if (((D_Dev < 0) && (D_Dev > (D_LReject_In * D_SDev))) || 
             ((D_Dev >= 0) && (D_Dev < (D_UReject_In * D_SDev)))){
-          D_A1_X.push_back(xNew[i_pos]);
+          D_A1_X.push_back(D_A1_X_In[i_pos]);
           D_A1_Y.push_back(D_A1_Y_In[i_pos]);
           if (B_HaveMeasureErrors)
-            D_A1_MeasureErrors.push_back(D_A1_MeasureErrorsBak[i_pos]);
-          I_A1_OrigPos.push_back(D_A1_Y_In[i_pos]);
+            V_MeasureErrors.push_back((*P_D_A1_MeasureErrors)[i_pos]);
+          I_A1_OrigPos.push_back(i_pos);
```

Comment:
I note for posterity that this is the sort of thing for which I'd expect the fix to be accompanied by a unit test. However, since we effectively don't have tests at this point (they are added by a future commit: 👍), we can let this slide.